docs/docs/20-usage/90-advanced-usage.md

```diff
@@ -89,7 +93,41 @@ steps:
      - <<: *post_cmds
 ```
 
-## References
+### References
+
+- [Official YAML specification](https://yaml.org/spec/1.2.2/#3222-anchors-and-aliases)
+- [YAML Cheatsheet](https://learnxinyminutes.com/docs/yaml)
+
+## Persisting environment data between steps
+
+One can create a file containing environment variables, and then source it in each step that needs them.
+
+```yml
+steps:
+  init:
+    image: bash
+    commands:
+      echo "FOO=hello" >> envvars
+      echo "BAR=world" >> envvars
+
+  debug:
+    image: bash
+    commands:
+      - source envvars
+      - echo $FOO
+```
+
+## Declaring global variables in `docker-compose.yml`
+
+As described in [Global environment variables](./50-environment.md#global-environment-variables), one can define global variables:
+
+```yml
+services:
+  woodpecker-server:
+    # ...
+    environment:
+      - WOODPECKER_ENVIRONMENT=first_var:value1,second_var:value2
+      # ...
+```
 
-- [Official specification](https://yaml.org/spec/1.2.2/#3222-anchors-and-aliases)
-- [Cheatsheet](https://learnxinyminutes.com/docs/yaml)
+Note that this tightly couples the server and app configurations (where the app is a completely separate application). But this is a good option for truly global variables which should apply to all steps in all pipelines for all apps.
```

Comment:
Actually not sure if we really need this part at all. Its somehow just a duplicate and pretty specific.